### PR TITLE
feat(protection): model mismatch

### LINKED ISSUE #854: [Bug]: vehicle type swap crashed
### Describe the bug

this is probably a crash sent by p2c using the local exchange model type

### Steps To Reproduce

.

### Expected Behavior

received_clone_create
received_clone_sync
no longer work

### Screenshots and/or Logs

```
<paste your logs here>
```
Dumping registers:RAX: 0x0RCX: 0x1B48C0DAE00RDX: 0x1RBX: 0x0RSI: 0x1B490FA7120RDI: 0x1RSP: 0xB286EFB100RBP: 0xB286EFB230R8:  0xFR9:  0x1R10: 0x7FF63752D310R11: 0xB286EFB280R12: 0x7FF635F11000R13: 0x3R14: 0x4000R15: 0x3stack dump [0]	 F:\steam\steamapps\common\Grand Theft Auto V\GTA5.exe+0xF4B6A9 0x7FF63651B6A9stack dump [1]	 F:\steam\steamapps\common\Grand Theft Auto V\GTA5.exe+0xFE6520 0x7FF6365B6520stack dump [2]	 F:\steam\steamapps\common\Grand Theft Auto V\GTA5.exe+0x1103CB3 0x7FF6366D3CB3stack dump [3]	 F:\steam\steamapps\common\Grand Theft Auto V\GTA5.exe+0x941106 0x7FF635F11106stack dump [4]	 F:\steam\steamapps\common\Grand Theft Auto V\GTA5.exe+0x163AAB9 0x7FF636C0AAB9stack dump [5]	 F:\steam\steamapps\common\Grand Theft Auto V\GTA5.exe+0x933BC9 0x7FF635F03BC9stack dump [6]	 F:\steam\steamapps\common\Grand Theft Auto V\GTA5.exe+0x950A85 0x7FF635F20A85stack dump [7]	 F:\steam\steamapps\common\Grand Theft Auto V\GTA5.exe+0x15F4C55 0x7FF636BC4C55stack dump [8]	 F:\steam\steamapps\common\Grand Theft Auto V\GTA5.exe+0x15F448B 0x7FF636BC448Bstack dump [9]	 F:\steam\steamapps\common\Grand Theft Auto V\GTA5.exe+0x27648 0x7FF6355F7648stack dump [10]	 F:\steam\steamapps\common\Grand Theft Auto V\GTA5.exe+0x20D37 0x7FF6355F0D37stack dump [11]	 F:\steam\steamapps\common\Grand Theft Auto V\GTA5.exe+0x15F4094 0x7FF636BC4094stack dump [12]	 F:\steam\steamapps\common\Grand Theft Auto V\GTA5.exe+0x1494 0x7FF6355D1494stack dump [13]	 F:\steam\steamapps\common\Grand Theft Auto V\GTA5.exe+0x12DD997 0x7FF6368AD997stack dump [14]	 F:\steam\steamapps\common\Grand Theft Auto V\GTA5.exe+0x17E678C 0x7FF636DB678Cstack dump [15]	 C:\Windows\System32\KERNEL32.DLL+0x155A0 BaseThreadInitThunkstack dump [16]	 C:\Windows\SYSTEM32\ntdll.dll+0x485B RtlUserThreadStartExiting after fatal event  (FATAL_EXCEPTION). Fatal type:  EXCEPTION_ACCESS_VIOLATIONLog content flushed successfully to sink

### Store

Steam

### Additional context

_No response_

### Build

- [X] I've confirmed the issue exists on the latest version of YimMenu

### PR DESCRIPTION
**Pull request by Aure7138**
Thanks to @sch-lda for helping to test

Closes #854

### CODE REVIEW

**Comment by maybegreat48** [comment]
This crash has existed for a long time, you should check objects, vehicles and doors too. Does the exception handler protect against this one?

File: src/util/model_info.hpp [hunk 1]
```diff
@@ -54,10 +54,15 @@ namespace big
 			return nullptr;
 		}
 
-		static bool is_model_of_type(const rage::joaat_t hash, const eModelType type)
+		template<typename T, typename ...Args>
+		static bool is_model_of_type(const rage::joaat_t hash, const T arg, const Args... args)
```
Comment (Yimura) [inline src/util/model_info.hpp:58]: Why is T a template argument? Why is `const T arg` never used?

Can you change `Args... args` to something more descriptive like `ModelTypes... types`?

**Comment by Aure7138** [comment]
> This crash has existed for a long time, you should check objects, vehicles and doors too. Does the exception handler protect against this one?

It just triggers the exception infinitely, I guess because the model is not deleted

File: src/util/model_info.hpp (hunk 1 above)
Comment (Aure7138) [reply src/util/model_info.hpp:58]: T exists to ensure at least one argument, since ...Args allows 0 arguments

Comment (Aure7138) [reply src/util/model_info.hpp:58]: You are right, I forgot to use arg, just changed

**Comment by Aure7138** [comment]
ready for review

**Review by Yimura — DISMISSED** [review]

**Review by Yimura — APPROVED** [review]

**Pull request merged**